File: frontend/src/types/message.ts
export type MessageRole = 'user' | 'assistant' | 'system';

export interface Message {
  id: string;
  role: MessageRole;
  content: string;
  timestamp: string;
  metadata?: {
    pipelineExecutionId?: string;
    consensusScore?: number;
    nodeResponses?: Array<{
      nodeId: string;
      status: string;
      output?: string;
    }>;
  };
}

export interface ChatRequest {
  message: string;
  pipelineId?: string;
  conversationId?: string;
}

export interface ChatResponse {
  id: string;
  message: Message;
  pipelineExecutionId: string;
  consensusScore?: number;
  nodeResponses?: Array<{
    nodeId: string;
    status: string;
    output?: string;
  }>;
}

File: backend/app/models/message.py
from enum import Enum
from typing import Optional
from pydantic import BaseModel, Field
from datetime import datetime
import uuid


class MessageRole(str, Enum):
    USER = "user"
    ASSISTANT = "assistant"
    SYSTEM = "system"


class Message(BaseModel):
    id: str = Field(default_factory=lambda: str(uuid.uuid4()))
    role: MessageRole
    content: str
    timestamp: datetime = Field(default_factory=datetime.utcnow)
    metadata: Optional[dict] = None


class ChatRequest(BaseModel):
    message: str
    pipeline_id: Optional[str] = None
    conversation_id: Optional[str] = None


class ChatResponse(BaseModel):
    id: str = Field(default_factory=lambda: str(uuid.uuid4()))
    message: Message
    pipeline_execution_id: str
    consensus_score: Optional[float] = None
    node_responses: Optional[list[dict]] = None
